src/worksheet.cc

```diff
@@ -223,3 +229,157 @@ Napi::Value Worksheet::WriteURL(const Napi::CallbackInfo& info) {
                       Format::Get(info[3]));
   return env.Undefined();
 }
+
+Napi::Value Worksheet::Autofilter(const Napi::CallbackInfo& info) {
+  auto env = info.Env();
+  worksheet_autofilter(worksheet,
+                       info[0].As<Napi::Number>(),
+                       info[1].As<Napi::Number>().Uint32Value(),
+                       info[2].As<Napi::Number>(),
+                       info[3].As<Napi::Number>().Uint32Value());
+  return env.Undefined();
+}
+
+Napi::Value Worksheet::DataValidationCell(const Napi::CallbackInfo& info) {
+  auto env = info.Env();
+  worksheet_data_validation_cell(
+      worksheet,
+      info[0].As<Napi::Number>(),
+      info[1].As<Napi::Number>().Uint32Value(),
+      info[2].As<Napi::External<lxw_data_validation>>().Data());
+  return env.Undefined();
+}
+
+Napi::Value Worksheet::DataValidationRange(const Napi::CallbackInfo& info) {
+  auto env = info.Env();
+
+  worksheet_data_validation_range(worksheet,
+                                  info[0].As<Napi::Number>(),
+                                  info[1].As<Napi::Number>().Uint32Value(),
+                                  info[2].As<Napi::Number>(),
+                                  info[3].As<Napi::Number>().Uint32Value(),
+                                  DataValidation(info[4]));
+
+  return env.Undefined();
+}
+
+Napi::Number callDateMethod(const Napi::Object& date, const char* method) {
+  return date.Get(method).As<Napi::Function>().Call({}).As<Napi::Number>();
+}
+
+lxw_datetime convertDate(const Napi::Object& date) {
+  return {
+      .year =
+          static_cast<int>(callDateMethod(date, "getFullYear").Uint32Value()),
+      .month =
+          static_cast<int>(callDateMethod(date, "getMonth").Uint32Value() + 1),
+      .day = static_cast<int>(callDateMethod(date, "getDate").Uint32Value()),
+      .hour = static_cast<int>(callDateMethod(date, "getHours").Uint32Value()),
+      .min = static_cast<int>(callDateMethod(date, "getMinutes").Uint32Value()),
+      .sec = callDateMethod(date, "getSeconds").Uint32Value() +
+             callDateMethod(date, "getMilliseconds").DoubleValue() / 1000.0,
+  };
+}
+
+DataValidation::DataValidation(const Napi::Value& value) {
+  const auto obj = value.As<Napi::Object>();
+
+  if (!obj.Get("validate").IsUndefined()) {
+    data_validation.validate =
+        obj.Get("validate").As<Napi::Number>().Uint32Value();
+  }
+  if (!obj.Get("criteria").IsUndefined()) {
+    data_validation.criteria =
+        obj.Get("criteria").As<Napi::Number>().Uint32Value();
+  }
+  if (!obj.Get("ignoreBlank").IsUndefined()) {
+    data_validation.ignore_blank =
+        obj.Get("ignoreBlank").As<Napi::Boolean>().Value();
```

Comment:
I don't think the `.Value()`, `.DoubleValue()`, `.c_str()`, etc. are required. They should be picked up implicitly.

Reply:
Removed `.Value()` and `.DoubleValue()`. `.c_str()` doesn't get picked up